### ISSUE TITLE
Inconsistent rendering of schemas describing the same data shape

### ISSUE THREAD

**Issue by llamamoray** [mozilla-services/react-jsonschema-form#604]
### Description

I've got two schemas that produce the same resulting shape of data from the form:
```
    const schema1 = {
      type: 'object',
      properties: {
        multipleChoicesList: {
          uniqueItems: true,
          items: {
            enum: ['Entity A', 'Entity B', 'Entity C', 'Entity D'],
            type: 'string',
          },
          type: 'array',
          title: 'A multiple choices list',
        },
      },
    };

    const schema2 = {
      definitions: {
        entityList: {
          enum: ['Entity A', 'Entity B', 'Entity C'],
          type: 'string',
        },
      },
      type: 'object',
      properties: {
        'Entity List': {
          uniqueItems: true,
          items: { $ref: '#/definitions/entityList' },
          type: 'array',
        },
      },
    };
```

They produce two different forms:

schema1:
![image](https://user-images.githubusercontent.com/2970147/26880417-4a12dd3c-4b8c-11e7-8718-d7c1b6e1903f.png)

schema2:
![image](https://user-images.githubusercontent.com/2970147/26880268-cc6a244e-4b8b-11e7-95fc-10cb283ea646.png)

Is this considered a bug in the library? It means that adding a uiSchema ` 'ui:widget': 'checkboxes' ` does not work with schema2.

### Steps to Reproduce

Apply schemas above to playground.

#### Expected behavior

Two schemas describing the same shape to produce the same form elements.

### Version

0.48.0

**Comment by glasserc** [comment]
Yes, that looks like a bug. Using `$ref` shouldn't change the schema.

**Comment by taroghion** [comment]
@glasserc Do you know where in the code the problem is?

**Comment by glasserc** [comment]
No, not off the top of my head, sorry.

**Comment by llamamoray** [comment]
@glasserc This seems to be fixed in the current live playground. Do you know when the next release will be?

**Comment by glasserc** [comment]
Yeah, this got fixed by #581. I just released v0.50.0 which should contain the fix.
